src/thread_id.rs

```diff
@@ -31,11 +30,14 @@ impl ThreadIdManager {
         if let Some(id) = self.free_list.pop() {
             id.0
         } else {
+            // we don't allow 1 before MAX to be returned because our buckets can only contain
+            // up to usize::MAX - 1 elements, but this shouldn't have any impact in practice.
+            if self.free_from >= usize::MAX - 1 {
+                panic!("Ran out of thread IDs");
+            }
```

Comment:
That is true but usize::MAX - 1 threads are possible theoretically speaking and as such i covered this case. If we only care about actually possible behavior i think a bunch of other theoretical safety measures in the crate can be removed as well, is this the path that is desired for this crate?

Reply:
No it is absolutely not possible since each thread consumes at least 2 bytes of memory.

What other safety measures do you have in mind? I suppose the array can be made a bit smaller since the maximum number of threads is capped.

Reply:
Okay, so because of the Thread struct there are actually a couple of bytes that need to be consumed by each thread, that is true. I would like to add a comment anyway. For other measures, it is possible to reduce the number of buckets to:
```
const BUCKETS: usize = (POINTER_WIDTH - POINTER_SIZE_BYTES.leading_zeros() as u8) as usize;
const POINTER_SIZE_BYTES: u8 = POINTER_WIDTH / 8;

```

Reply:
But with the information regarding the Thread struct, we could make it even smaller, to be precise
```
const BUCKETS: usize = (POINTER_WIDTH - (POINTER_SIZE_BYTES + size_of::<Thread>()).leading_zeros() as u8) as usize;
```

Reply:
I would rather keep things simple: just delete this check but keep the bucket array as it is. I know it's a bit arbitrary but I don't want to get into the weeds of trying to guess how much memory a thread uses on any particular platform.

Reply:
Fixed.